core/domain/collection_domain.py

```diff
@@ -1148,10 +1148,10 @@ def get_skill_id_from_skill_name(self, skill_name):
                 present.
         """
         skill_id = None
-        if any([
-                skill_name == skill.name
-                for skill in self.skills.itervalues()]):
-            skill_id = skill.id
+        skill_list = [skill_name == skill.name
```

Comment:
I'd like to suggest a refactor -- can we just use a for loop? I think it's easy to understand that way.

Reply:
Yaa it would be easy. As when it gets matching value them it will return the id. I need to ask that there can exists skills with same name.

Reply:
Nope. All names are different. You can break as soon as you get to the first match.

Reply:
I am thinking of some thing like this: 
```py
        for skill in self.skills.itervalues():
            if skill_name == skill.name:
                return skill.id
        return None
```

Reply:
Views @seanlip

Reply:
In the future, please just go ahead and do the change and submit it for review. However, I can confirm that your code snippet looks good. The only change I would make is to set skill_id = skill.id inside the for loop and then break, and only return skill_id at the very end, so that the whole function has just one return statement which works correctly in all the cases.

Reply:
Sure @seanlip

Reply:
Will mind that in future.